### PR TITLE
Fix bank_index Parameter Override

### PR DESCRIPTION
**Pull request by Depetrol**
This PR solves #2410 by allowing explicitly set `bank_index` parameter to override in Python target.

### CODE REVIEW

File: test/Cpp/src/ActionWithNoReaction.lf [hunk 1]
```diff
@@ -11,7 +11,7 @@ reactor foo {
   logical action a
 
   reaction(x) -> y, a {=
-    y.set(2*(*x.get()));
+    y.set(2*(*x.undefined_name15291838()));
```
Comment (edwardalee) [inline test/Cpp/src/ActionWithNoReaction.lf:14]: This looks like an unintended and unrelated change?

**Review by edwardalee — APPROVED** [review]
Good catch!  LGTM, except that you seem to have an unintended and unrelated change.

File: test/Cpp/src/ActionWithNoReaction.lf (hunk 1 above)
Comment (Depetrol) [reply test/Cpp/src/ActionWithNoReaction.lf:14]: Yes this is unintended, probably has something to do with the code editor. I'll remove it.

**Review by lhstrh — APPROVED** [review]
